### PR TITLE
Adding isCleanUp to Transaction

### PR DESCRIPTION
**Pull request by tinystarinagalaxy**

### CODE REVIEW

File: proto/api.proto [hunk 1]
```diff
@@ -610,6 +610,11 @@ message PostCancelAllRequest {
   repeated string openOrdersAddresses = 3;
 }
 
+message Transaction {
```
Comment (aspin) [inline proto/api.proto]: @DimaCantemir suggestion:

```
message TransactionMessage {
    string content = 1;
    bool isCleanup = 2;
}
```

**Review by aspin — APPROVED** [review]
thoughts on if you want to make this universal, including endpoints that only return a single transaction? I'm a little mixed on the consistency of all endpoints vs. teh useless data fields

File: proto/api.proto (hunk 1 above)
Comment (tinystarinagalaxy) [reply proto/api.proto]: done

**Comment by tinystarinagalaxy** [comment]
> thoughts on if you want to make this universal, including endpoints that only return a single transaction? I'm a little mixed on the consistency of all endpoints vs. the useless data fields

what specific endpoint you have in mind?  I checked all other endpoints. they look clean to me so far. there is some validity with the consistency point that you are raising. 

if it looks awkwardly inconsistent, we can apply the same change later as well. it's a minor change.

**Comment by aspin** [comment]
> > thoughts on if you want to make this universal, including endpoints that only return a single transaction? I'm a little mixed on the consistency of all endpoints vs. the useless data fields
> 
> what specific endpoint you have in mind? I checked all other endpoints. they look clean to me so far. there is some validity with the consistency point that you are raising.
> 
> if it looks awkwardly inconsistent, we can apply the same change later as well. it's a minor change.

I was thinking all of them, for example 

```
message PostOrderResponse {
  string transaction = 1;
  string openOrdersAddress = 2;
}

```

I'm leaning a bit more to consistency of fields vs. the unnecessary ones, but could be swayed either way. any thoughts @DimaCantemir @kdonthi @atulsriv @yobloxroute ?

**Comment by atulsriv** [comment]
> > > thoughts on if you want to make this universal, including endpoints that only return a single transaction? I'm a little mixed on the consistency of all endpoints vs. the useless data fields
> > 
> > 
> > what specific endpoint you have in mind? I checked all other endpoints. they look clean to me so far. there is some validity with the consistency point that you are raising.
> > if it looks awkwardly inconsistent, we can apply the same change later as well. it's a minor change.
> 
> I was thinking all of them, for example
> 
> ```
> message PostOrderResponse {
>   string transaction = 1;
>   string openOrdersAddress = 2;
> }
> ```
> 
> I'm leaning a bit more to consistency of fields vs. the unnecessary ones, but could be swayed either way. any thoughts @DimaCantemir @kdonthi @atulsriv @yobloxroute ?

I think consistency of fields looks a bit cleaner. Not sure how much it really matters vs. keeping the unnecessary fields, but it looks better imo.

**Comment by tinystarinagalaxy** [comment]
> PostOrderResponse

hmm, that would be a bigger change, let's create a ticket for it, if we decide to go for all.

**Comment by tinystarinagalaxy** [comment]
> > PostOrderResponse
> 
> hmm, that would be a bigger change, let's create a ticket for it, if we decide to go for all.

i merge this for now

**Pull request merged**